lib/loader/loader.js

```diff
@@ -217,6 +217,11 @@ var loader, define, requireModule, require, requirejs;
   };
 
   define = function(name, deps, callback) {
+    var module = registry[name];
+    if (module && module.state !== 'new') {
+      return;
```

Comment:
Can you also please put in a comment here so that it is easier to reference later why we do this only for everything except `new`?